cmd/translate/main_grpc_integration_test.go

```diff
@@ -157,102 +180,106 @@ func createService(ctx context.Context, t *testing.T) *translatev1.Service {
 func Test_UploadTranslationFile_gRPC(t *testing.T) {
 	t.Parallel()
 
-	ctx := context.Background()
-
-	service := createService(ctx, t)
-
-	// Requests
+	ctx, spanEnd := trace(context.Background(), t)
+	// Using t.Cleanup instead of defer to ensure that the sub-tests are finished before closing parent span.
+	t.Cleanup(spanEnd)
 
-	happyRequest := randUploadRequest(t, service.Id)
-
-	invalidArgumentMissingLangRequest := randUploadRequest(t, service.Id)
-	invalidArgumentMissingLangRequest.Language = ""
-
-	notFoundServiceIDRequest := randUploadRequest(t, gofakeit.UUID())
-
-	tests := []struct {
+	type test struct {
 		request      *translatev1.UploadTranslationFileRequest
 		name         string
 		expectedCode codes.Code
-	}{
-		{
-			name:         "Happy path",
-			request:      happyRequest,
-			expectedCode: codes.OK,
-		},
-		{
-			name:         "Invalid argument missing language",
-			request:      invalidArgumentMissingLangRequest,
-			expectedCode: codes.InvalidArgument,
-		},
-		{
-			name:         "Not found service ID",
-			request:      notFoundServiceIDRequest,
-			expectedCode: codes.NotFound,
-		},
 	}
 
-	for _, tt := range tests {
-		tt := tt
-		t.Run(tt.name, func(t *testing.T) {
-			t.Parallel()
+	var tests []test
 
-			_, err := client.UploadTranslationFile(ctx, tt.request)
+	// Prepare
+	t.Run("Prepare tests", func(t *testing.T) {
```

Comment:
wrapping preparations for tests in subtest .e.g in this case inserting service for translation file, this way cleaner in Jaeger
<img width="1445" alt="image" src="https://github.com/translate-agent/translate/assets/69954124/68869873-2c43-4881-83c1-98c34cb84294">

Reply:
nevermind... it was bad idea :D